modules/jaggeryapps/ml/site/analysis/analysis.jag

```diff
@@ -268,7 +268,46 @@
         $(document).on('click', '#create-model', function(e){ 
             e.preventDefault();
 
-            window.location.href = '../model/model.jag?analysisId='+analysisId+'&datasetId='+datasetId+'&projectName='+projectName+'&analysisName='+analysisName+"&fromModels=true";
+            var algorithmName, algorithmType;
+
+            $.ajax({
+                type: "GET",
+                url: serverUrl+"/api/analyses/"+analysisId+"/algorithmName",
+                async : false,
+                dataType: 'text',
+                beforeSend : function(xhr) {
+                    xhr.setRequestHeader("Authorization", "Basic " + authEncoded);
+                },
+                success : function(res){
+                    algorithmName = res;
+                },
+                error : function(res){
+                    if (crtValArea.prev("ctrl-validation-panel").length == 0){
+                        crtValArea.before(buildNotifications(res.responseJSON.status));
+                    }
+                }
+            });
+
+            $.ajax({
+                type: "GET",
+                url: serverUrl+"/api/analyses/"+analysisId+"/algorithmType",
+                async : false,
+                dataType: 'text',
+                beforeSend : function(xhr) {
+                    xhr.setRequestHeader("Authorization", "Basic " + authEncoded);
+                },
+                success : function(res){
+                    algorithmType = res;
+                },
+                error : function(res){
+                    if (crtValArea.prev("ctrl-validation-panel").length == 0){
+                        crtValArea.before(buildNotifications(res.responseJSON.status));
```

Comment:
Can you please fix the argument to buildNotifications s.t. it passes res.responseText ? Otherwise back-end error message is not getting appeared in the UI.